runtime/commonMain/src/kotlinx/serialization/Descriptors.kt

```diff
@@ -21,20 +21,37 @@ sealed class PrimitiveKind: SerialKind() {
     object FLOAT : PrimitiveKind()
     object DOUBLE : PrimitiveKind()
     object CHAR : PrimitiveKind()
-    object STRING: PrimitiveKind()
+    object STRING : PrimitiveKind()
 }
 
-sealed class StructureKind: SerialKind() {
-    object CLASS: StructureKind()
-    object LIST: StructureKind()
-    object MAP: StructureKind()
+sealed class StructureKind : SerialKind() {
+    object CLASS : StructureKind()
+    object LIST : StructureKind()
+    object MAP : StructureKind()
 }
 
-sealed class UnionKind: SerialKind() {
-    object OBJECT: UnionKind()
-    object ENUM_KIND: UnionKind() // https://github.com/JetBrains/kotlin-native/issues/1447
-    object SEALED: UnionKind()
-    object POLYMORPHIC: UnionKind()
+@Suppress("unused", "PropertyName")
+sealed class UnionKind : SerialKind() {
+    object OBJECT : UnionKind()
+    object ENUM_KIND : UnionKind() // https://github.com/JetBrains/kotlin-native/issues/1447
+
+    @Deprecated(
+        "Moved out from UnionKind to simplify instance check for both POLYMORPHIC and SEALED. You can use 'is PolymorphicKind' now.",
+        ReplaceWith("PolymorphicKind.OPEN"),
+        DeprecationLevel.ERROR
+    )
+    val POLYMORPHIC = PolymorphicKind.OPEN
+    @Deprecated(
+        "Moved out from UnionKind to simplify instance check for both POLYMORPHIC and SEALED. You can use 'is PolymorphicKind' now.",
+        ReplaceWith("PolymorphicKind.SEALED"),
+        DeprecationLevel.ERROR
+    )
+    val SEALED = PolymorphicKind.SEALED
```

Comment:
Why 'nor source code migration'?

Reply:
Because it's properties in the sealed class, not in its companion.
You can access them only with a specific instance (e.g. `UnionKind.OBJECT.SEALED`)

Reply:
Still not addressed

Reply:
Ah, I finally get it. I'd suggest to add `companion object` to `UnionKind` so migration will work.